sdk/storage/azblob/client.go

```diff
@@ -150,6 +139,25 @@ func (c *Client) Download(ctx context.Context, containerName string, blobName st
 	return c.svc.NewContainerClient(containerName).NewBlobClient(blobName).Download(ctx, o.BlobOptions)
 }
 
+// DownloadToWriterAt downloads an Azure blob to a WriterAt in parallel.
+// Offset and count are optional, pass 0 for both to download the entire blob.
+func (c *Client) DownloadToWriterAt(ctx context.Context, containerName string, blobName string, offset, count int64, writer io.WriterAt, o *DownloadToWriterAtOptions) error {
+	return c.svc.NewContainerClient(containerName).NewBlobClient(blobName).DownloadToWriterAt(ctx, offset, count, writer, o)
+}
+
+// DownloadToBuffer downloads an Azure blob to a buffer with parallel.
+// Offset and count are optional, pass 0 for both to download the entire blob.
+func (c *Client) DownloadToBuffer(ctx context.Context, containerName string, blobName string, offset, count int64, _bytes []byte, o *DownloadToBufferOptions) error {
+	return c.svc.NewContainerClient(containerName).NewBlobClient(blobName).DownloadToBuffer(ctx, offset, count, shared.NewBytesWriter(_bytes), o)
```

Comment:
`DownloadToBuffer` should be returning `(int, error)` so that callers can get the number of bytes that were downloaded (can be fixed in a separate PR).